src/routes/publish.js

```diff
@@ -118,21 +118,34 @@ export function publishRoutes() {
     // Koa which will expose it appropriately to the caller.
     await Promise.all([pipelinePromise, jsonCheckerPromise]);
 
+    // Step 2: Upload the data to Google Cloud Storage.
+    // All data chunks have been stored in the concatener, that we'll read from
+    // to feed the google write stream.
     const storage = gcsStorageCreate(config);
     const hash = hasherTransform.sha1();
-
     const googleStorageStream = storage.getWriteStreamForFile(hash);
 
-    // We can't use Stream.finished here because of a problem with Google's
+    // We don't use `pipeline` because we want to cleanly unpipe and cleanup
+    // when the request is aborted.
+    concatener.pipe(googleStorageStream);
+
+    ctx.req.on('aborted', () => {
+      log.debug('request-aborted', 'The request has been aborted!');
+      concatener.unpipe(googleStorageStream);
+      const error = new BadRequestError('The request has been aborted.');
+      googleStorageStream.destroy(error);
+      concatener.destroy(error);
+    });
```

Comment:
From my test this doesn't seem to make a big difference. But this does no harm 🤷